src/base/settingvalue.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2016  Vladimir Golovnev <glassez@yandex.ru>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#ifndef SETTINGVALUE_H
+#define SETTINGVALUE_H
+
+#include <functional>
+#include <QString>
+
+#include "settingsstorage.h"
+
+template <typename T>
+class CachedSettingValue
+{
+    using ProxyFunc = std::function<T (const T&)>;
+
+public:
+    explicit CachedSettingValue(const char *keyName, const T &defaultValue = T()
+            , ProxyFunc proxyFunc = [](const T &value) { return value; })
```

Comment:
I haven't read the whole diff so I might not get the whole picture: It seems that you want to pass a lambda as a parameter. Discussion on this topic results in 2 possibilities:
1. Making a template out of the function which takes a lambda
2. Using std::function

However, you do both here. Are you sure you can't just drop std::function here?
Relevant SO topic with examples: https://stackoverflow.com/questions/16111285/how-to-pass-and-execute-anonymous-function-as-parameter-in-c11